### PR TITLE
Ofir/aws support all regions

### PR DESCRIPTION
**Pull request by itsofirblink**

### CODE REVIEW

File: implementation/implementation.go [hunk 1]
```diff
@@ -1746,6 +1746,37 @@ func (p *AWSPlugin) GetActions() []plugin.Action {
 	return p.actions
 }
 
+func executeAction(packageName string, ctx *plugin.ActionContext, actionExecutor ActionExecutor, actionParameters map[string]interface{}) (map[string]interface{}, error) {
+	if actionParameters["awsRegion"] == "*" {
```
Comment (danielsblinkops) [inline implementation/implementation.go:1750]: const it already lol

File: implementation/implementation.go [hunk 2]
```diff
@@ -1746,6 +1746,37 @@ func (p *AWSPlugin) GetActions() []plugin.Action {
...
+		result := make(map[string]interface{})
+		availableRegions := awsutil.GetServiceRegions(packageName)
+		for _, region := range availableRegions {
+			actionParameters["awsRegion"] = region
```
Comment (danielsblinkops) [inline implementation/implementation.go:1754]: I think its time to write a wrapper with getters andsetters

File: implementation/implementation.go (hunk 1 above)
Comment (danielsblinkops) [inline implementation/implementation.go:1750]: Move this to a func

File: implementation/implementation.go [hunk 3]
```diff
@@ -1746,6 +1746,37 @@ func (p *AWSPlugin) GetActions() []plugin.Action {
...
+			regionResult, err := executeAction(packageName, ctx, actionExecutor, actionParameters)
+			if err != nil {
+				result[region] = map[string]interface{}{
+					"error": err.Error(),
+				}
```
Comment (danielsblinkops) [inline implementation/implementation.go:1759]: dont else here

continue inside the if

File: implementation/implementation.go [hunk 4]
```diff
@@ -1746,6 +1746,37 @@ func (p *AWSPlugin) GetActions() []plugin.Action {
...
+			} else {
+				result[region] = regionResult
+			}
+		}
```
Comment (danielsblinkops) [inline implementation/implementation.go:1763]: new fuckingline

File: aws/awsutil/blinkutil.go [hunk 5]
```diff
@@ -250,4 +251,24 @@ func UnpackParameters(parameters map[string]interface{}, shape interface{}) map[
...
+		operationServiceRegions = operationService.Regions()
+	}
+
+	if operationServiceRegions != nil {
+		for _, region := range operationServiceRegions {
+			operationRegions = append(operationRegions, region.ID())
+		}
+	}
+
+	return operationRegions
```
Comment (danielsblinkops) [inline aws/awsutil/blinkutil.go:273]: return an empty slice and not a nil one, even beter a clear error

**Review by danielsblinkops — CHANGES_REQUESTED** [review]

File: aws/awsutil/blinkutil.go (hunk 5 above)
Comment (itsofirblink) [reply aws/awsutil/blinkutil.go:273]: empty slice it is

**Pull request merged**